service/gcs/bridge/bridge.go

```diff
@@ -334,21 +334,37 @@ func (b *bridge) runExternalProcess(message []byte) (*prot.ContainerExecuteProce
 	return response, nil
 }
 
-func (b *bridge) waitOnProcess(message []byte, header *prot.MessageHeader) (*prot.ContainerWaitForProcessResponse, error) {
-	response := &prot.ContainerWaitForProcessResponse{MessageResponseBase: newResponseBase()}
+func (b *bridge) waitOnProcess(message []byte, header *prot.MessageHeader) (response *prot.ContainerWaitForProcessResponse, err error) {
```

Comment:
The issue is that in theory, if RegisterExitHook fails, it could still register the exit hook and thus two responses (both an error and the one specified by the exit hook) could be sent. This would never occur if RegisterExitHook behaves properly, but I put this change in mainly to not send nonsense back to the HCS if it does not behave properly. I'm fine with not making this change, however, especially if your async bridge changes remove the necessity for it.